ballerina/tests/from_xml_test.bal

```diff
@@ -0,0 +1,448 @@
+    var x1 = xml `<!-- outer comment -->`;
+    var x2 = xml `<name>0.5</name>`;
+    xml x3 = x1 + x2;
+
+    map<decimal> expected = {
+        name: 0.5
+    };
+
+    map<decimal> actual = check fromXml(x3);
+    test:assertEquals(actual, expected, msg = "testXmlToMapJson1 result incorrect");
+}
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testXmlToMapFloat1() returns error? {
+    var x1 = xml `<!-- outer comment -->`;
+    var x2 = xml `<name>0.5</name>`;
+    xml x3 = x1 + x2;
+
+    map<float> expected = {
+        name: 0.5
+    };
+
+    map<float> actual = check fromXml(x3);
+    test:assertEquals(actual, expected, msg = "testXmlToMapJson1 result incorrect");
+}
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testXmlToMapXml1() returns error? {
+    var x1 = xml `<name>0.5</name>`;
+
+    map<xml> expected = {
+        "#content": xml `<name>0.5</name>`
+    };
+
+    map<xml> actual = check fromXml(x1);
+    test:assertEquals(actual, expected, msg = "testXmlToMapJson1 result incorrect");
+}
+
+xml xmlData = xml `<Invoice xmlns="example.com" attr="attr-val" xmlns:ns="ns.com" ns:attr="ns-attr-val">
+                        <PurchesedItems>
+                            <PLine><ItemCode>223345</ItemCode><Count>10</Count></PLine>
+                            <PLine><ItemCode>223300</ItemCode><Count>7</Count></PLine>
+                            <PLine><ItemCode discount="22%">200777</ItemCode><Count>7</Count></PLine>
+                        </PurchesedItems>
+                        <Address xmlns="">
+                            <StreetAddress>20, Palm grove, Colombo 3</StreetAddress>
+                            <City>Colombo</City>
+                            <Zip>00300</Zip>
+                            <Country>LK</Country>
+                        </Address>
+                    </Invoice>`;
+
+@test:Config {
+    groups: ["fromXml"]
+}
+function testToJsonWithComplexXmlElement() returns Error? {
+    map<json> j = check fromXml(xmlData);
+    map<json> expectedOutput = {
+        Invoice: {
+            PurchesedItems: {
+                PLine: [
+                    {ItemCode: "223345", Count: "10"},
+                    {ItemCode: "223300", Count: "7"},
+                    {
+                        ItemCode: {"discount": "22%", "#content": "200777"},
+                        Count: "7"
+                    }
+                ]
+            },
+            Address: {
+                StreetAddress: "20, Palm grove, Colombo 3",
+                City: "Colombo",
+                Zip: "00300",
+                Country: "LK",
+                "xmlns": ""
+            },
+            "xmlns:ns": "ns.com",
+            "xmlns": "example.com",
+            "attr": "attr-val",
+            "ns:attr": "ns-attr-val"
+        }
+    };
+    test:assertEquals(j, expectedOutput, msg = "testToJsonComplexXmlElement result incorrect");
+}
+
+type Commercial4 record {
```

Comment:
Why do we need to wrap with this another record?

Can't we do it like,
```
type BookStore record {
   string storeName;
   int postalCode;
   ....
   Address address;
}

type Address record {
   string street;
   string city;
   string country;
}
```

Reply:
Updated in [dfe5e7d](https://github.com/ballerina-platform/module-ballerina-xmldata/pull/431/commits/dfe5e7dedb5743868cfea01e85d934cd4c7faaa5)